src/cli/trn_cli_common.c

```diff
@@ -617,3 +617,64 @@ int trn_cli_parse_ebpf_prog_stage(const cJSON *jsonobj,
 	}
 	return 0;
 }
+
+int trn_cli_parse_network_policy_cidr(const cJSON *jsonobj,
+				      struct rpc_trn_vsip_cidr_t *cidrval)
+{
+	cJSON *prefixlen = cJSON_GetObjectItem(jsonobj, "prefixlen");
```

Comment:
prefixlen field seems better to be derived based on cidr subnet part. Suggest to consider changing the protocol to include only tunnel-id, local-ip, cidr-ip, cidr-subnet-prefixlen.

Reply:
Do you mean just change the naming from "prefixlen" to "cidr-subnet-prefixlen"?

Reply:
sort-of; what I suggested is to calculate the final prefixlen in transit daemon code, instead of relying on operator code. The rational is operator should not know about the detail; it knows only the destination CIDR (in your words cidr-subnet-prefixlen).